### ISSUE TITLE
Endless recursion with AttachContext

### ISSUE THREAD

**Issue by rantav** [savsgio/atreugo#65]
I'm sorry, but there seems to be an issue with the new `AttachContext` function. 

When you attach a context that contains the original request context and then call `Value` with a key that does not exist, you get an endless recursion (stack overflow). 

Example: 

```go
func Test_ValueEndlessRecursion(t *testing.T) {
	ctx := acquireRequestCtx(new(fasthttp.RequestCtx))
	ctx.AttachContext(context.WithValue(ctx, "yyy", "zzz"))
	ctx.Value("xxx")
}
```

Or even a simpler example: 
```go
func Test_ValueEndlessRecursion(t *testing.T) {
	ctx := acquireRequestCtx(new(fasthttp.RequestCtx))
	ctx.AttachContext(ctx)
	ctx.Value("xxx")
}
```

Result: 

```
=== RUN   Test_ValueEndlessRecursion
runtime: goroutine stack exceeds 1000000000-byte limit
fatal error: stack overflow

runtime stack:
runtime.throw(0x14f95c7, 0xe)
	/usr/local/go/src/runtime/panic.go:774 +0x72
runtime.newstack()
	/usr/local/go/src/runtime/stack.go:1046 +0x6e9
runtime.morestack()
	/usr/local/go/src/runtime/asm_amd64.s:449 +0x8f

goroutine 50 [running]:
runtime.getitab(0x14834c0, 0x1498000, 0x1, 0xc0001d8030)
	/usr/local/go/src/runtime/iface.go:33 +0x37b fp=0xc0201dc358 sp=0xc0201dc350 pc=0x100ae6b
runtime.assertE2I2(0x14834c0, 0x1498000, 0xc0001d8000, 0x1498000, 0xc0001d8000, 0x0)
	/usr/local/go/src/runtime/iface.go:487 +0x43 fp=0xc0201dc388 sp=0xc0201dc358 pc=0x100bfe3
github.com/savsgio/atreugo/v10.(*RequestCtx).AttachedContext(0xc0001d6000, 0x14478c0, 0x15873d0)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:64 +0x77 fp=0xc0201dc3c8 sp=0xc0201dc388 pc=0x13e5c97
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:79 +0x64 fp=0xc0201dc400 sp=0xc0201dc3c8 pc=0x13e5d44
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc438 sp=0xc0201dc400 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc470 sp=0xc0201dc438 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc4a8 sp=0xc0201dc470 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc4e0 sp=0xc0201dc4a8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc518 sp=0xc0201dc4e0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc550 sp=0xc0201dc518 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc588 sp=0xc0201dc550 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc5c0 sp=0xc0201dc588 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc5f8 sp=0xc0201dc5c0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc630 sp=0xc0201dc5f8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc668 sp=0xc0201dc630 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc6a0 sp=0xc0201dc668 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc6d8 sp=0xc0201dc6a0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc710 sp=0xc0201dc6d8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc748 sp=0xc0201dc710 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc780 sp=0xc0201dc748 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc7b8 sp=0xc0201dc780 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc7f0 sp=0xc0201dc7b8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc828 sp=0xc0201dc7f0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc860 sp=0xc0201dc828 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc898 sp=0xc0201dc860 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc8d0 sp=0xc0201dc898 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc908 sp=0xc0201dc8d0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc940 sp=0xc0201dc908 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc978 sp=0xc0201dc940 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dc9b0 sp=0xc0201dc978 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dc9e8 sp=0xc0201dc9b0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dca20 sp=0xc0201dc9e8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dca58 sp=0xc0201dca20 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dca90 sp=0xc0201dca58 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcac8 sp=0xc0201dca90 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcb00 sp=0xc0201dcac8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcb38 sp=0xc0201dcb00 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcb70 sp=0xc0201dcb38 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcba8 sp=0xc0201dcb70 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcbe0 sp=0xc0201dcba8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcc18 sp=0xc0201dcbe0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcc50 sp=0xc0201dcc18 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcc88 sp=0xc0201dcc50 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dccc0 sp=0xc0201dcc88 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dccf8 sp=0xc0201dccc0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcd30 sp=0xc0201dccf8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcd68 sp=0xc0201dcd30 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcda0 sp=0xc0201dcd68 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcdd8 sp=0xc0201dcda0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dce10 sp=0xc0201dcdd8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dce48 sp=0xc0201dce10 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dce80 sp=0xc0201dce48 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dceb8 sp=0xc0201dce80 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcef0 sp=0xc0201dceb8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcf28 sp=0xc0201dcef0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcf60 sp=0xc0201dcf28 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dcf98 sp=0xc0201dcf60 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dcfd0 sp=0xc0201dcf98 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd008 sp=0xc0201dcfd0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd040 sp=0xc0201dd008 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd078 sp=0xc0201dd040 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd0b0 sp=0xc0201dd078 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd0e8 sp=0xc0201dd0b0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd120 sp=0xc0201dd0e8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd158 sp=0xc0201dd120 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd190 sp=0xc0201dd158 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd1c8 sp=0xc0201dd190 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd200 sp=0xc0201dd1c8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd238 sp=0xc0201dd200 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd270 sp=0xc0201dd238 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd2a8 sp=0xc0201dd270 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd2e0 sp=0xc0201dd2a8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd318 sp=0xc0201dd2e0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd350 sp=0xc0201dd318 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd388 sp=0xc0201dd350 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd3c0 sp=0xc0201dd388 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd3f8 sp=0xc0201dd3c0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd430 sp=0xc0201dd3f8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd468 sp=0xc0201dd430 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd4a0 sp=0xc0201dd468 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd4d8 sp=0xc0201dd4a0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd510 sp=0xc0201dd4d8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd548 sp=0xc0201dd510 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd580 sp=0xc0201dd548 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd5b8 sp=0xc0201dd580 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd5f0 sp=0xc0201dd5b8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd628 sp=0xc0201dd5f0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd660 sp=0xc0201dd628 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd698 sp=0xc0201dd660 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd6d0 sp=0xc0201dd698 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd708 sp=0xc0201dd6d0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd740 sp=0xc0201dd708 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd778 sp=0xc0201dd740 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd7b0 sp=0xc0201dd778 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd7e8 sp=0xc0201dd7b0 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd820 sp=0xc0201dd7e8 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd858 sp=0xc0201dd820 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd890 sp=0xc0201dd858 pc=0x13e5d71
context.(*valueCtx).Value(0xc0001d8000, 0x14478c0, 0x15873d0, 0x0, 0x0)
	/usr/local/go/src/context/context.go:520 +0x58 fp=0xc0201dd8c8 sp=0xc0201dd890 pc=0x10ee038
github.com/savsgio/atreugo/v10.(*RequestCtx).Value(0xc0001d6000, 0x14478c0, 0x15873d0, 0xc0001d8000, 0x1)
	/Users/ran/dev/src/github.com/savsgio/atreugo/context.go:80 +0x91 fp=0xc0201dd900 sp=0xc0201dd8c8 pc=0x13e5d71
...additional frames elided...
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0001c4100, 0x14fe808, 0x1a, 0x1514110, 0x1085601)
	/usr/local/go/src/testing/testing.go:961 +0x377
testing.runTests.func1(0xc0001c4000)
	/usr/local/go/src/testing/testing.go:1202 +0x78
testing.tRunner(0xc0001c4000, 0xc0001bddc0)
	/usr/local/go/src/testing/testing.go:909 +0xc9
testing.runTests(0xc0000b8b80, 0x18b56c0, 0x26, 0x26, 0x0)
	/usr/local/go/src/testing/testing.go:1200 +0x2a7
testing.(*M).Run(0xc000126200, 0x0)
	/usr/local/go/src/testing/testing.go:1117 +0x176
main.main()
	_testmain.go:124 +0x135

goroutine 34 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:144 +0x96
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:29 +0x41

Process finished with exit code 1
```

**Comment by rantav** [comment]
The second case (`ctx.AttachContext(ctx)`) can be solved by checking if the accepted `ctx` is the same as the method receiver (and if so, noop). 
The first case (`ctx.AttachContext(context.WithValue(ctx, "yyy", "zzz"))`) is tougher, I don't have a good lead how to solve it...

**Comment by savsgio** [comment]
Hi @rantav,

I will investigate about that to see how to fix it.

Thanks for your issue.

**Comment by rantav** [comment]
Hi man, just checking, any progress on that?

**Comment by savsgio** [comment]
Hi, Yes i'm studing the best solution.

Meanwhile, create a new context with value from a `context.Background()` instead of a `RequestCtx`, because i'm afraid it's so difficult to avoid the loop (stack overflow) if use directly the own `RequestCtx` due to overwrite the `RequestCtx.Value()`

So in your case, do:
```go
newCtx, span = trace.StartSpan(context.Background(), name)
ctx.AttachContext(newCtx)

// After middleware or view
trace.FromContext(ctx.AttachedContext())
```

Instead of
```go
newCtx, span = trace.StartSpan(ctx, name)
ctx.AttachContext(newCtx)

trace.FromContext(ctx)
```

I'm thinking, maybe the best solution is used directly the attached context for create a new context instead of use `RequestCtx` and attach it again to the `RequestCtx`, like `net/http`. Because i could check the new context before attach it but It is very expensive and slow with `reflect`

And adds a "WARNING" for not use the RequestCtx for creates a new context from it and attach them again to itself. But it's so confuse 😥

**Comment by savsgio** [comment]
Hi @rantav ,

Yes, i'm studing the best solution, don't worry I expect to have it soon.

**Comment by savsgio** [comment]
Fixed! 🎉 

Please, read the added documentation!

**Comment by rantav** [comment]
Thanks man, that was unexpectedly complicated...
Did you create a new release?

**Comment by savsgio** [comment]
Yes, I will release a new version today or tomorrow!

**Comment by savsgio** [comment]
Released v10.2.0 🎉
